File: docs/archive/ScotlandsMountains.Api/Loader/Models/Mountain.cs
using System.Collections.Generic;

namespace ScotlandsMountains.Api.Loader.Models
{
    public class Mountain : Entity
    {
        public Mountain() : base(PartitionKeyFrom.Id) { }

        public string Name { get; set; }

        public List<string> Aliases { get; set; }

        public Location Location { get; set; }
        
        public int DobihId { get; set; }
        
        public string GridRef { get; set; }
        
        public Height Height { get; set; }
        
        public Prominence Prominence { get; set; }
        
        public string Features { get; set; }
        
        public string Observations { get; set; }

        public MountainSummary Parent { get; set; }

        public RegionSummary Region { get; set; }

        public CountySummary County { get; set; }

        public IList<ClassificationSummary> Classifications { get; set; } = new List<ClassificationSummary>();

        public IList<MapSummary> Maps { get; set; } = new List<MapSummary>();
    }

    public class MountainSummary : Summary
    {
        public MountainSummary(Mountain mountain)
            : base(mountain)
        {
            Name = mountain.Name;
            Location = mountain.Location;
            Height = mountain.Height;
        }

        public string Name { get; set; }

        public Location Location { get; set; }

        public Height Height { get; set; }
    }
}

File: docs/archive/ScotlandsMountains.UI/src/app/Models/Mountain.ts
import { Classification } from "./Classification";

export interface Mountain {
    id: string;
    position?: number;
    name: string;
    aliases?: string[];
    location: Location;
    dobihId?: number;
    gridRef?: string;
    height: Height;
    prominence?: Prominence;
    features?: string;
    observations?: string;
    parent?: Mountain;
    region?: Region;
    county?: County;
    classifications?: Classification[];
    maps?: Map[];
}

export interface Location {
    type: string;
    coordinates: number[];
}

export interface Height {
    metres: number;
}

export interface Prominence
{
    metres: number;
    measuredFrom: string;
    measuredFromHeight: Height;
}

export interface Region {
    id: string;
    name: string;
}

export interface County {
    id: string;
    name: string;
}

export interface Map {
    id: string;
    code: string;
    scale: number;
}